pkg/controller/endpointslice/reconciler.go

```diff
@@ -287,6 +290,13 @@ func (r *reconciler) reconcileByPortMapping(
 			}
 		}
 
+		svcLabelsUpdated := !utillabels.Contains(existingSlice.Labels, service.Labels)
+		if svcLabelsUpdated {
+			for label, val := range service.Labels {
+				existingSlice.Labels[label] = val
+			}
+		}
```

Comment:
This should follow the pattern in EndpointSliceMirroring controller, need to be careful to ensure that labels on Service can't override important EndpointSlice labels: https://github.com/kubernetes/kubernetes/blob/90723ebbae0753a08a34aa10e7bc5efc98b2cf5a/pkg/controller/endpointslicemirroring/utils.go#L100-L105

Reply:
done